### DISCUSSION TITLE
Redux devtools time travelling issue

### DISCUSSION THREAD

**Discussion by stropho** [pmndrs/zustand#2878] [Bug report]
### Bug Description

Hi, I have an issue with time traveling in redix devtools.
I am fairly confident it worked earlier this year (but I wouldn't bet my life on it 😉 )

The way I understand the problem which imho appeared out of nowhere: redux devtools use a stringified JSON state snapshot and use that for time travelling (my naive assumption). And such stringification removes all methods such as on Map or Set or even "actions" contained in the store - that makes the time travel useless.

As an example I found an existing codesandbox https://mr54pq.csb.app/
As soon as you try to time travel (= e.g. jump to an older action ), the app breaks as it tries to read from a Map in the store.

The sandbox uses zustand 3.5.4, I also tried 4.4.7 and 5.0.1
For the Redux Devtools I use Chrome extension version 3.2.7, but also tried older version 3.0.19 from early 2023. Still the same...

Am I doing something wrong? Is there a workaround for it? Or is it a super new things that nobody has noticed yet?

### Reproduction Link

https://mr54pq.csb.app/

**Comment by dai-shi** [comment]
Can anyone help look into it?
